src/System.Diagnostics.Contracts/tests/BasicContractsTests.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.Contracts;
+using Xunit;
+
+namespace System.Diagnostics.Contracts.Tests
+{
+    public class BasicContractsTest
+    {
+        [Fact]
+        public static void AssertTrueDoesNotRaiseEvent()
+        {
+            bool eventRaised = false;
+            Contract.ContractFailed += (s, e) =>
+            {
+                eventRaised = true;
+                e.SetHandled();
+            };
+
+            Contract.Assert(true);
+            Assert.False(eventRaised, "ContractFailed event was raised");
+        }
+
+        [Fact]
+        public static void AssertFalseRaisesEvent()
+        {
+            bool eventRaised = false;
+            Contract.ContractFailed += (s, e) =>
+            {
+                eventRaised = true;
+                e.SetHandled();
+            };
+
+            Contract.Assert(false);
```

Comment:
Good catch - I updated the test to verify the behavior based on its configuration.  Our build infrastructure already has the configuration (debug, release) of the test match the configuration of the runtime (where this is implemented, in mscorlib).